analyzers/tests/SonarAnalyzer.UnitTest/TestCases/Hotspots/ConfiguringLoggers_AspNetCore6.cs

```diff
@@ -0,0 +1,100 @@
+using System.Collections.Generic;
+using Microsoft.AspNetCore;
+using Microsoft.Extensions.Options;
+using Microsoft.AspNetCore;
+using Microsoft.AspNetCore.Builder;
+using Microsoft.AspNetCore.Hosting;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Logging;
+using Microsoft.AspNetCore.Http;
+using Microsoft.Extensions.Configuration;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Hosting;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Logging.Console;
+using Microsoft.Extensions.Logging.AzureAppServices;
+using Microsoft.Extensions.Logging.EventLog;
+using Microsoft.Extensions.Logging.EventSource;
+
+
+namespace MvcApp
+{
+    //RSPEC S4792: https://jira.sonarsource.com/browse/RSPEC-4792
+    public class ProgramLogging
+    {
+        public static IWebHostBuilder CreateWebHostBuilder(string[] args) =>
+            WebHost.CreateDefaultBuilder(args) // Noncompliant {{Make sure that this logger's configuration is safe.}}
+                .ConfigureLogging((hostingContext, logging) =>
+                {
+                    // ...
+                })
+                .UseStartup<StartupLogging>();
+    }
+
+
+    public class StartupLogging
+    {
+        public IServiceCollection ConfigureServices(IServiceCollection services)
+        {
+            services.AddLogging(logging => // Noncompliant {{Make sure that this logger's configuration is safe.}}
+            {
+                logging.AddConsole(); // Noncompliant
+                // ...
+            });
+            return services;
+        }
+
+        public void Configure(IApplicationBuilder app)
+        {
+            IConfiguration config = null;
+            LogLevel level = LogLevel.Critical;
+            bool includeScopes = false;
+            Func<string, LogLevel, bool> filter = null;
+            EventLogSettings eventLogSettings = null;
+
+            using (var loggerFactory = LoggerFactory.Create(builder => builder.AddAzureWebAppDiagnostics() // Noncompliant [1,2,3,4,5,6,7]
+                                                                              .AddAzureWebAppDiagnostics(azureSettings => { azureSettings.IncludeScopes = true; }) // Every invocation in this chain is noncompliant
+                                                                              .AddConsole(options => { options.FormatterName = "simple"; })
+                                                                              .AddDebug()
+                                                                              .AddEventLog()
+                                                                              .AddEventLog(eventLogSettings)
+                                                                              .AddEventSourceLogger()
+                                                                              .AddFilter(filter))) // Filter cannot be given to AddConsole as input. Is this an FN?
```

Comment:
Note: I'm not sure an issue should be raised also for `AddFilter`.
In previous versions of apsnetcore the `filter` input could be given to `AddConsole` but now it seems that this is possible only through `AddFilter`.

Reply:
If I understand correctly the concern in this rule is with adding loggers. All the previous methods will create a logger (e.g. [AddDebug](https://docs.microsoft.com/en-us/dotnet/api/microsoft.extensions.logging.debugloggerfactoryextensions.adddebug?view=dotnet-plat-ext-6.0#microsoft-extensions-logging-debugloggerfactoryextensions-adddebug(microsoft-extensions-logging-iloggingbuilder)) ). 

Since AddFilter is only concerned with filtering and no logger is created, it should not be marked as vulnerable.

I think it would be better to document its behavior on a separate line.

Reply:
I'm not sure what you mean regarding documenting the behavior of `AddFilter`.

I added one separate line in the code though to make clear that it's compliant.

Reply:
This was it :)